src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -155,6 +155,15 @@ def _txt2obj_gc(backend, name):
     return obj
 
 
+def _encode_ocsp_nocheck(backend, ext):
+    """
+    The OCSP No Check extension is defined as a null ASN.1 value. We can just
+    return that value directly here in the pp, r tuple form the other
+    extension encoding functions use.
```

Comment:
Yeah, that would be clearer.

On Sun, Aug 9, 2015 at 11:50 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/backend.py
> <https://github.com/pyca/cryptography/pull/2233#discussion_r36589695>:
>
> > @@ -155,6 +155,15 @@ def _txt2obj_gc(backend, name):
> >      return obj
> >
> >
> > +def _encode_ocsp_nocheck(backend, ext):
> > +    """
> > +    The OCSP No Check extension is defined as a null ASN.1 value. We can just
> > +    return that value directly here in the pp, r tuple form the other
> > +    extension encoding functions use.
>
> We can build a null by doing some magic with ASN1_TYPE but since an ASN.1
> null is always \x05\x00 I'm not in favor of calling (several) C functions
> to get those two bytes.
>
> The comment says "return that value directly here in the pp, r tuple form"
> so it isn't saying it won't return a tuple. Would it be more clear if it
> read "Rather than calling OpenSSL we can return a Python bytestring value
> in a list"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2233/files#r36589695>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084